plugins/topology/common/components/Status.tsx

```diff
@@ -0,0 +1,169 @@
+ * @param {string} [popoverTitle] - (optional) title for popover
+ * @param {ReactNode} [children] - (optional) children for the component
+ * @example
+ * ```tsx
+ * <Status status='Warning' />
+ * ```
+ */
+const Status: React.FC<StatusProps> = ({
+  status,
+  title,
+  iconOnly,
+  noTooltip,
+  className,
+  children,
+}) => {
+  const statusProps = {
+    title: title || status,
+    iconOnly,
+    noTooltip,
+    className,
+  };
+  switch (status) {
+    case 'New':
+      return (
+        <StatusIconAndText {...statusProps} icon={<HourglassStartIcon />} />
+      );
+
+    case 'Pending':
+      return (
+        <StatusIconAndText {...statusProps} icon={<HourglassHalfIcon />} />
+      );
+
+    case 'Planning':
+      return (
+        <StatusIconAndText {...statusProps} icon={<ClipboardListIcon />} />
+      );
+
+    case 'ContainerCreating':
+    case 'UpgradePending':
+    case 'PendingUpgrade':
+    case 'PendingRollback':
+      return <StatusIconAndText {...statusProps} icon={<InProgressIcon />} />;
+
+    case 'In Progress':
+    case 'Installing':
+    case 'InstallReady':
+    case 'Replacing':
+    case 'Running':
+    case 'Updating':
+    case 'Upgrading':
+    case 'PendingInstall':
+      return <StatusIconAndText {...statusProps} icon={<SyncAltIcon />} />;
+
+    case 'Cancelled':
+    case 'Deleting':
+    case 'Expired':
+    case 'Not Ready':
+    case 'Cancelling':
+    case 'Terminating':
+    case 'Superseded':
+    case 'Uninstalling':
+      return <StatusIconAndText {...statusProps} icon={<BanIcon />} />;
+
+    case 'Warning':
+    case 'RequiresApproval':
+      return (
+        <StatusIconAndText
+          {...statusProps}
+          icon={<YellowExclamationTriangleIcon />}
+        />
+      );
+
+    case 'ContainerCannotRun':
+    case 'CrashLoopBackOff':
+    case 'Critical':
+    case 'ErrImagePull':
+    case 'Error':
+    case 'Failed':
+    case 'Failure':
+    case 'ImagePullBackOff':
+    case 'InstallCheckFailed':
+    case 'Lost':
+    case 'Rejected':
+    case 'UpgradeFailed':
+      return (
+        <StatusIconAndText {...statusProps} icon={<RedExclamationCircleIcon />}>
+          {children}
+        </StatusIconAndText>
+      );
+
+    case 'Accepted':
+    case 'Active':
+    case 'Bound':
+    case 'Complete':
+    case 'Completed':
+    case 'Created':
+    case 'Enabled':
+    case 'Succeeded':
+    case 'Ready':
+    case 'Up to date':
+    case 'Loaded':
+    case 'Provisioned as node':
+    case 'Preferred':
+    case 'Connected':
+    case 'Deployed':
+      return (
+        <StatusIconAndText {...statusProps} icon={<GreenCheckCircleIcon />} />
+      );
+
+    case 'Info':
+      return (
+        <StatusIconAndText {...statusProps} icon={<BlueInfoCircleIcon />}>
+          {children}
+        </StatusIconAndText>
+      );
+
+    case 'Unknown':
+      return <StatusIconAndText {...statusProps} icon={<UnknownIcon />} />;
+
+    case 'PipelineNotStarted':
```

Comment:
is this needed? take a look at other status as well and good to have one which makes sense